File: cuarray/samples/cpp/src/CuArray_sort.cpp
#include <cuarray.h>
#include <random>
#include <iostream>

int main() {
    std::cout
            << "Running "
            << __FILE__
            << std::endl;
/* Creates a new float CuArray instance */
    CuArray<float> *cuArray = new CuArray<float>();

/* Initialize the CuArray with 300 rows and 300 columns */
    auto rows = 300;
    auto cols = 300;
    cuArray->init(rows,
                  cols);

/* Fill the CuArray with random values */
    for (int i = 0; i < cuArray->m(); i++) {
        for (int j = 0; j < cuArray->n(); j++) {
            cuArray->host()[i * cuArray->n() + j] =
                    static_cast<float>(rand() / (float) RAND_MAX);
        }
    }

/* Create a new CuArray that contains the sorted data from the
 * 8th row of the original CuArray. */
    auto sortedCuArray = cuArray->sort(7);

/* Print the sorted CuArray. */
    for (int j = 0; j < sortedCuArray->n(); j++) {
        std::cout
                << sortedCuArray->get(0,
                                      j)
                << std::endl;
    }

/* Cleanup time. */
    delete cuArray;
    delete sortedCuArray;
    return 0;
}

File: cuarray/samples/python/CuArray_sort.py
import numpy as np

"""
Always precede CuArray with the data type
Here we are importing the CuArray float template 
"""
from cuarray import FloatCuArray

print("Running", __file__)

"""
Create a new float CuArray instance 
"""
float_cuarray = FloatCuArray()

"""
Create a random float32 numpy array with 10 rows
and 10 columns
"""
numpy_array = np.random.rand(10, 10).astype(np.float32)

"""Load the numpy array into the CuArray"""
float_cuarray.fromNumpy2D(numpy_array)

"""
Perform an out of place descending sort on the 
8th column of float_cuarray
"""
sorted_cuarray = float_cuarray.sort(7)

"""
Print the 8th row of the original 
CuArray and sorted_cuarray
"""
print(sorted_cuarray)
print(float_cuarray[7])